gef-ui/server/controllers/application-details/index.js

```diff
@@ -199,9 +200,27 @@ const applicationDetails = async (req, res, next) => {
   let url;
   const link = `/gef/application-details/${dealId}`;
 
+  if (!getUserAuthorisationLevelsToApplication(user)) {
```

Comment:
- Because your PR just adds redirect, not sure you need to introduce new access check.
- function `getUserAuthorisationLevelsToApplication` is already called in this file bellow. Not sure if this function is used here correctly, as second parameter is not provided.

Reply:
I added this condition as the test `redirects to dashboard if user is not authorised` is expecting a condition and redirect if not authorised. I tried to add this condition with i`f (!application || !getUserAuthorisationLevelsToApplication(user)`) but the functionality breaks when user clicks on `View Unissued Facilities` link.

Reply:
I removed function `getUserAuthorisationLevelsToApplication`